src/account/tests.py

```diff
@@ -1,3 +1,243 @@
+
+    # -------------------------------------------------------------------------
+    # SINGLE view
+    # -------------------------------------------------------------------------
+
+    # -------------------------------------------------------------------------
+    # GET
+    # -------------------------------------------------------------------------
+
+    def test_get_single_random_admin(self):
+        self.client.force_login(User.objects.get(username='admin'))
+        get_idx = Profile.objects.order_by('?').first().pk
+        url = ''.join([reverse('profile-list'), str(get_idx), '/'])
+        response = self.client.get(url)
+
+        profile = Profile.objects.get(pk=get_idx)
+        serializer = ProfileSerializer(profile)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, serializer.data)
+
+        self.client.logout()
+
+    def test_get_single_random_mod(self):
+        self.client.force_login(User.objects.get(username='mod'))
+        get_idx = Profile.objects.order_by('?').first().pk
+        url = ''.join([reverse('profile-list'), str(get_idx), '/'])
+        response = self.client.get(url)
+
+        profile = Profile.objects.get(pk=get_idx)
+        serializer = ProfileSerializer(profile)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, serializer.data)
+
+        self.client.logout()
+
+    def test_get_single_own_user(self):
+        self.client.force_login(User.objects.get(username='user'))
+        get_idx = User.objects.get(username='user').pk
+        url = ''.join([reverse('profile-list'), str(get_idx), '/'])
+        response = self.client.get(url)
+
+        profile = Profile.objects.get(pk=get_idx)
+        serializer = ProfileSerializer(profile)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, serializer.data)
+
+        self.client.logout()
+
+    def test_get_single_not_own_user(self):
+        self.client.force_login(User.objects.get(username='user'))
+        get_idx = User.objects.get(username='other_user').pk
+        url = ''.join([reverse('profile-list'), str(get_idx), '/'])
+        response = self.client.get(url)
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+        self.client.logout()
+
+    def test_get_single_non_exist_admin(self):
+        self.client.force_login(User.objects.get(username='admin'))
+        get_idx = Profile.objects.all().count() + 1
+        url = ''.join([reverse('profile-list'), str(get_idx), '/'])
+        response = self.client.get(url)
+
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+        self.client.logout()
+
+    # -------------------------------------------------------------------------
+    # PUT
+    # -------------------------------------------------------------------------
+
+    def test_put_single_admin(self):
```

Comment:
Couldn't write proper test for PUT. When sending the request, set content-type to multipart/form-data, because it is needed for uploading an image, but multipart doesn't support nested objects, so I tried to fixed it, not done yet. The error I'm getting at the moment: `{'user': [ErrorDetail(string='This field is required.', code='required')], 'avatar': [ErrorDetail(string='The submitted file is empty.', code='empty')]}`